docs/spec/v1beta1/imageupdateautomations.md

```diff
@@ -441,6 +464,100 @@ spec:
+characteristics necessitate a few workarounds. Let's take a look at an example
+configuration that plays nicely with Gerrit:
+
+```yaml
+spec:
+  git:
+    checkout:
+      ref:
+        branch: main
+    commit:
+      author:
+        email: flux@localdomain
+        name: flux
+      messageTemplate: |
+        Perform automatic image update
+
+        Automation name: {{ .AutomationObject }}
+
+        Files:
+        {{ range $filename, $_ := .Updated.Files -}}
+        - {{ $filename }}
+        {{ end }}
+        Objects:
+        {{ range $resource, $_ := .Updated.Objects -}}
+        - {{ $resource.Kind }} {{ $resource.Name }}
+        {{ end }}
+        Images:
+        {{ range .Updated.Images -}}
+        - {{ . }}
+        {{ end }}
+        {{- $ChangeId := .AutomationObject -}}
+        {{- $ChangeId = printf "%s%s" $ChangeId ( .Updated.Files | toString ) -}}
+        {{- $ChangeId = printf "%s%s" $ChangeId ( .Updated.Objects | toString ) -}}
+        {{- $ChangeId = printf "%s%s" $ChangeId ( .Updated.Images | toString ) }}
+        Change-Id: {{ printf "I%s" ( sha256sum $ChangeId | trunc 40 ) }}
+    push:
+      branch: auto
+      refspec: refs/heads/auto:refs/heads/main
+```
+
+This tells image-automation-controller to clone the repo using the `main` branch
+but, run its update logic and make any commits using the provided message
+template on the `auto` branch. It pushes them to the `auto` branch and then,
+pushes the `HEAD` of the `auto` branch to the `HEAD` of the remote `main`
+branch. The message template ensures that we introduce a [Change-Id](https://gerrit-review.googlesource.com/Documentation/concept-changes.html#change-id)
+at the bottom of the commit message.
+
+The reason behind pushing the commit to a branch initially is to avoid multiple
+[Patch Sets](https://gerrit-review.googlesource.com/Documentation/concept-patch-sets.html).
+If we omitted `push.branch` and just specified
+`push.refspec: refs/heads/main:refs/heads/main`, then we would be able to open
+a [Change](https://gerrit-review.googlesource.com/Documentation/concept-changes.html)
+as intended. But, every time the reconciler does a fresh clone of the `main`
+branch while the Change is still open, it'll run its update logic against the
+files in the `main` branch resulting in a new commit being pushed with the same
+changes to the already open Change. Specifying a `push.branch` lets us work
+around this since, it tells the reconciler to run the update logic against
+the `auto` branch, which already contains the desired commit. This pattern
+is also recommended in the [Gerrit docs](https://gerrit-review.googlesource.com/Documentation/intro-gerrit-walkthrough-github.html#create-change).
+
+Another thing to note here is the syntax of `push.refspec`. Instead of it being
+`HEAD:refs/for/main`, which is what most Gerrit users use, we specify the full
+refname `refs/heads/auto` in the source part of the refpsec.
```

Comment:
```suggestion
Another thing to note is the syntax of `.push.refspec`. Instead of it being
`HEAD:refs/for/main`, commonly used by Gerrit users, we specify the full
refname `refs/heads/auto` in the source part of the refpsec.
```

Readability suggestion.